### ISSUE TITLE
[Usability] User experiences an unexpected interaction with a component.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#80110]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** Benefits Non-Disability
**Product:** Burial Form 530
**Feature:** 21P-530 Parity Updates

## Findings details
**VA.gov Experience Standard - issue:** User experiences an unexpected interaction with a component.
**VA.gov Experience Standard - category:** Usability
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

On [v2 /separation-documents/upload](https://staging.va.gov/burials-and-memorials-v2/application/530/military-history/separation-documents/upload), [v2 /additional-information/death-certificate](https://staging.va.gov/burials-and-memorials-v2/application/530/additional-information/death-certificate), [v2 /additional-information/transportation-receipts](https://staging.va.gov/burials-and-memorials-v2/application/530/additional-information/transportation-receipts), [v2 /additional-information/additional-evidence](https://staging.va.gov/burials-and-memorials-v2/application/530/additional-information/additional-evidence), keyboard focus goes to the top of the page after uploading your first document. 

As a keyboard-only user, when focus goes somewhere unexpected you have to do a lot of extra tabbing to get back to the right place. As a screen reader user, focus determines what the next thing is that's announced by the screen reader, and hearing something unexpected from a different part of the page can be confusing and might lead you to believe you've been removed from the workflow. 

When uploading additional documents, keyboard focus seems to go to an element not visible on the page but it's still in the right neighborhood and recoverable after one tab.

<details><summary>Link, screenshot or steps to recreate</summary>

Using a focus tracking bookmarklet like [What has focus?](https://codepen.io/svinkle/pen/WgYRxq) I'm seeing:
- After the first file upload, focus goes to the "An official website of the United States government" button to toggle the USA government banner.
- After each subsequent file upload, focus goes to a `va-button` element with class `retry-upload`, which I don't see on the page.

</details>

## Recommended action

If possible, I think the expected behavior after a successful upload would be for focus to go to the file name for the file just uploaded, set using `tabindex="-1"`. That will announce the file name of the file just uploaded to screen reader users (informing them that the file upload was successful), make the delete button the next tab stop (easy error recovery if you uploaded the wrong thing), and puts you more or less back in the place you were before you uploaded the file.

## References

- **WCAG Success Criteria:** N/A
- **Modality:** Keyboard
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by Midge-dev** [comment]
@shiragoodman I know you're unlikely to be the reviewer who flagged this issue, but this seems closely related to issue [#19688](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19688). We've been successful in testing some code that was written specifically for this issue for our use case, however the focus ends up on the Add another button.

cc: @tblackwe @steele-lm

**Comment by shiragoodman** [comment]
Thanks @Midge-dev ! I am tagging @briandeconinck for his opinion. He was the reviewer who flagged this issue :)

**Comment by Midge-dev** [comment]
@briandeconinck in other applications the Add another file button is focused after upload and delete. Do we want the same interactions for consistency? Or do we want the focus on the aforementioned file name?

cc: @tblackwe @steele-lm 

[Screencast from 2024-04-09 11-38-25.webm](https://github.com/department-of-veterans-affairs/va.gov-team/assets/57480791/1e9cc575-ab06-4c2f-9245-e2cb1f54482e)

**Comment by briandeconinck** [comment]
Hi @Midge-dev (and @DLarson-Oddball)! Sorry for taking so long to get back to you on this --- I wanted to test a bunch of forms to make sure I don't lead you down the wrong path.

The short answer: It turns out we aren't super consistent across VA.gov on this kind of behavior! But moving focus to the next element (eg. the Add Another button) is definitely the more common pattern, and I think it makes sense to match that unless/until there's design system guidance saying otherwise. Please feel free to move forward with your solution matching that.

**Comment by DLarson-Oddball** [comment]
Thanks for your feedback @briandeconinck! 

@steele-lm code for this issue is now merged.
